File: MPI/mpicheck/tests/functions/2.0/MPI_Grequest_start.f90

        
subroutine var_0(val, ierr)
    integer val, ierr
end subroutine

subroutine var_1(val, ierr)
    integer val, ierr
end subroutine

subroutine var_2(val, ierr)
    integer val, ierr
end subroutine
        program main
        use mpi
        
        EXTERNAL var_0
       EXTERNAL var_1
       EXTERNAL var_2
       INTEGER(KIND=MPI_ADDRESS_KIND) var_3
       INTEGER var_4
       INTEGER var_5
        call mpi_grequest_start(var_0, var_1, var_2, var_3, var_4, var_5)
       call pmpi_grequest_start(var_0, var_1, var_2, var_3, var_4, var_5)
        end program main
    

File: MPI/mpicheck/tests/functions/2.0/MPI_Grequest_start.c
#include <mpi.h>
int main(char argc, char**argv)
{
    /* vars */
    MPI_Grequest_query_function *var_0;
    MPI_Grequest_free_function *var_1;
    MPI_Grequest_cancel_function *var_2;
    void *var_3;
    MPI_Request *var_4;
    int ret;
    /* calls */
    ret = MPI_Grequest_start(var_0, var_1, var_2, var_3, var_4);
    ret = PMPI_Grequest_start(var_0, var_1, var_2, var_3, var_4);
    return 0;
}
